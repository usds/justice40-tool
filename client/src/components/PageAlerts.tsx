import * as React from 'react';
import {Alert} from '@trussworks/react-uswds';
import {FormattedMessage} from 'gatsby-plugin-intl';

interface IPageAlertProps {
  betaWarning?: boolean;
  dataSourceWarning?: boolean;
}

export const PageAlerts = ({
  betaWarning = true,
  dataSourceWarning = false}: IPageAlertProps) => {
  return (
    <>
      { betaWarning ?
        <Alert className={'j40-sitealert'} type="info">
          <b><FormattedMessage
            id='header.alertTitleBeta'
            description={'Alerts that appear on every page - title'}
            defaultMessage={`Public beta`}/> - </b>
          <FormattedMessage
            id='header.alertBodyBeta'
            description={'Alerts that appear on every page'}
            defaultMessage={`This website will be continuously updated`}/>
          <br/>
        </Alert> :
        <></>
      }
      { dataSourceWarning ? // default is to not show
        <Alert className={'j40-sitealert'} type="warning">
          <b><FormattedMessage
            id='header.alertTitleLimitedData'
            description={'Alerts that appear on maps page - title'}
            defaultMessage={`Limited data sources`}/> â€” </b>
          <FormattedMessage
            id='header.alertBodyLimitedData'
            description={'Alerts that appear on maps page'}
            defaultMessage={`Datasets may be added, updated, or removed.`}/>
          <br/>
        </Alert> :
        <></>
      }
    </>
  );
};
