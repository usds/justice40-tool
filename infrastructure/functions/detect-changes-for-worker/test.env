REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1