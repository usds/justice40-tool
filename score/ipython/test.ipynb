{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
