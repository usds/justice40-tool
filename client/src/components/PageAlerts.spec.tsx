import React from 'react';
import {render} from '@testing-library/react';
import {PageAlerts} from './PageAlerts';
import {LocalizedComponent} from '../test/testHelpers';

describe('PageAlert', () => {
  const alertBetaTitle = `Public beta`;
  const alertDataLimitTitle = `Limited data sources`;

  it('renders correctly', () => {
    const {asFragment} = render(<LocalizedComponent><PageAlerts/></LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders only info', async () => {
    const {baseElement} = await render(
        <LocalizedComponent>
          <PageAlerts/>
        </LocalizedComponent>);
    // WARNING:
    // The following line is standard BUT DOES NOT WORK. I think it has to do
    // with jest and intl issues. `baseElement.textContent` is a work around
    // expect(await screen.getAllByText(alertBetaTitle)).toBeInTheDocument();

    const textElement = baseElement.textContent;
    expect(textElement).toContain(alertBetaTitle);
    expect(textElement).not.toContain(alertDataLimitTitle);
  });

  it('renders only warning', async () => {
    const {baseElement} = render(
        <LocalizedComponent>
          <PageAlerts betaWarning={false} dataSourceWarning={true}/>
        </LocalizedComponent>);
    const textElement = baseElement.textContent;
    expect(textElement).not.toContain(alertBetaTitle);
    expect(textElement).toContain(alertDataLimitTitle);
  });

  it('renders both', async () => {
    const {baseElement} = render(
        <LocalizedComponent>
          <PageAlerts betaWarning={true} dataSourceWarning={true}/>
        </LocalizedComponent>);
    const textElement = baseElement.textContent;
    expect(textElement).toContain(alertBetaTitle);
    expect(textElement).toContain(alertDataLimitTitle);
  });
});
