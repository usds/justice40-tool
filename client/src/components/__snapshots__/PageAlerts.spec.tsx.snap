// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAlert renders correctly 1`] = `
<DocumentFragment>
  <span
    class="j40-sitealerts-component"
  >
    <div
      class="usa-alert usa-alert--info j40-sitealert"
      data-testid="alert"
    >
      <div
        class="usa-alert__body"
      >
        <p
          class="usa-alert__text"
        >
          <b>
            Public beta - 
          </b>
          This website will be continuously updated
          <br />
        </p>
      </div>
    </div>
  </span>
</DocumentFragment>
`;
